internal/dao/stubs/pubkey_dao.go

```diff
@@ -64,9 +65,11 @@ func (stub *pubkeyDaoStub) Update(ctx context.Context, pubkey *models.Pubkey) er
 }
 
 func (stub *pubkeyDaoStub) GetById(ctx context.Context, id int64) (*models.Pubkey, error) {
+	var resultPk models.Pubkey
 	for _, pk := range stub.store {
 		if pk.AccountID == ctxAccountId(ctx) && pk.ID == id {
-			return pk, nil
+			resultPk = *pk // shallow copy of the struct
+			return &resultPk, nil
```

Comment:
Why do you do that?

Reply:
This is a shallow copy: https://go.dev/play/p/0aatJFCKdnW

Reply:
because otherwise during the test we would change the same structure as we are adding to the array.
So you add stubbed pubkey with name "Pubkey", run test that is expected to change the name, but the name is changes to the initial structure in the test also.
Which is unexpected behaviour IMHO.